kaldb/src/main/java/com/slack/kaldb/logstore/opensearch/OpenSearchAdapter.java

```diff
@@ -103,194 +83,118 @@
-    MapperService mapperService = buildMapperService(indexSettings, similarityService);
+  private final MapperService mapperService;
 
-    // todo - see SchemaAwareLogDocumentBuilderImpl.getDefaultLuceneFieldDefinitions
-    //  this needs to be adapted to include other field types once we have support
-    for (Map.Entry<String, LuceneFieldDef> entry : chunkSchema.entrySet()) {
-      try {
-        if (entry.getValue().fieldType == FieldType.LONG) {
-          registerField(
-              mapperService,
-              entry.getValue().name,
-              b -> b.field("type", entry.getValue().fieldType.name));
-        } else if (entry.getValue().fieldType == FieldType.STRING) {
-          registerField(mapperService, entry.getValue().name, b -> b.field("type", "keyword"));
-        }
-      } catch (Exception e) {
-        LOG.error("Error parsing schema mapping for {}", entry.getValue().toString(), e);
-      }
-    }
+  private final Map<String, LuceneFieldDef> chunkSchema;
 
+  public OpenSearchAdapter(Map<String, LuceneFieldDef> chunkSchema) {
+    IndexSettings indexSettings = buildIndexSettings();
+    SimilarityService similarityService = new SimilarityService(indexSettings, null, emptyMap());
+    this.mapperService = buildMapperService(indexSettings, similarityService);
     this.queryShardContext =
         buildQueryShardContext(
             KaldbBigArrays.getInstance(), indexSettings, similarityService, mapperService);
+    this.chunkSchema = chunkSchema;
   }
 
-  /** Serializes InternalAggregation to byte array for transport */
-  public static byte[] toByteArray(InternalAggregation internalAggregation) {
-    if (internalAggregation == null) {
-      return new byte[] {};
-    }
-
-    byte[] returnBytes;
-    try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
-      try (StreamOutput streamOutput = new OutputStreamStreamOutput(byteArrayOutputStream)) {
-        InternalAggregations internalAggregations =
-            new InternalAggregations(List.of(internalAggregation), null);
-        internalAggregations.writeTo(streamOutput);
+  /**
+   * Builds a Lucene query using the provided arguments, and the currently loaded schema. Uses the
+   * Opensearch Query String builder. TODO - use the dataset param in building query
+   *
+   * @see https://opensearch.org/docs/latest/query-dsl/full-text/query-string/
+   * @see
+   *     https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html
+   */
+  public Query buildQuery(
+      String dataset, String queryStr, long startTimeMsEpoch, long endTimeMsEpoch)
+      throws IOException {
+    LOG.trace("Query raw input string: '{}'", queryStr);
+    try {
+      Query query;
+      if (queryStr != null
+          && !queryStr.isEmpty()
+          && !queryStr.equals("*:*")
+          && !queryStr.equals("*")) {
+        QueryStringQueryBuilder queryStringQueryBuilder =
```

Comment:
Is there a query parser we can leverage to generate the `queryStr`?

Reply:
Good call - I had wanted to revisit this approach and forgot. 

I've refactored this so it doesn't use the query string in the same manner now. Instead of constructing a query string with all of the clauses and providing this to the string builder, I construct a bool query that combines a range query and the result of the query string builder with just the user's input string.